hr_employee_calendar_planning/models/hr_employee.py

```diff
@@ -113,6 +114,15 @@ def regenerate_calendar(self):
         for item in self:
             item._regenerate_calendar()
 
+    def copy(self, default=None):
+        self.ensure_one()
+        new = super().copy(default)
+        # We define the first as the main calendar
+        new.resource_id.calendar_id = fields.first(new.calendar_ids).calendar_id
```

Comment:
Can we avoid this? We are regenerating after the main calendar, and it's too late for filling the required field. I'm also doubting if the regenerate call is needed.

Reply:
I have tried other possibilities and this is the "_least hard_".

Maybe another option would be that `calendar_id` field of `resource.resource` is not copy, do you want me to investigate a bit more here?

Reply:
OK, if that's the reason, then please clarify it with a good comment. The current one is not clarifying anything. If I'm understanding OK the problem, the comment would be something like `# Define a good main calendar for being able to regenerate it later`.